File: frontend/src/types/market.ts
export interface MarketPriceItem {
  crop: string;
  variety?: string | null;
  market: string;
  district?: string | null;
  state?: string | null;
  price: number;
  min_price?: number | null;
  max_price?: number | null;
  modal_price?: number | null;
  unit: string;
  date?: string | null;
  source?: string | null;
}

export interface PriceSummary {
  lowest_price: number;
  highest_price: number;
  average_price: number;
  unit: string;
  total_records: number;
}

export interface MarketPriceResponse {
  success: boolean;
  configured: boolean;
  status: 'success' | 'no_data' | 'unconfigured' | 'error';
  source: string;
  last_updated?: string | null;
  selected_crop?: string | null;
  selected_location?: string | null;
  results: MarketPriceItem[];
  summary?: PriceSummary | null;
  message?: string | null;
}

File: backend/app/schemas/market.py
from typing import List, Optional
from pydantic import BaseModel

class MarketPriceItem(BaseModel):
    crop: str
    variety: Optional[str] = None
    market: str
    district: Optional[str] = None
    state: Optional[str] = None
    price: float
    min_price: Optional[float] = None
    max_price: Optional[float] = None
    modal_price: Optional[float] = None
    unit: str = "Rs/Quintal"
    date: Optional[str] = None
    source: Optional[str] = None

class PriceSummary(BaseModel):
    lowest_price: float
    highest_price: float
    average_price: float
    unit: str = "Rs/Quintal"
    total_records: int

class MarketPriceResponse(BaseModel):
    success: bool
    configured: bool
    status: str  # "success" | "no_data" | "unconfigured" | "error"
    source: str
    last_updated: Optional[str] = None
    selected_crop: Optional[str] = None
    selected_location: Optional[str] = None
    results: List[MarketPriceItem] = []
    summary: Optional[PriceSummary] = None
    message: Optional[str] = None
